### PR TITLE
🐛 OSIDB-3440: Fix tracker links

### PR DESCRIPTION
**Pull request by C-Valen**
# OSIDB-3440 Fix tracker links

## Checklist:

- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Hot fix for wrong links on trackers table.

## Changes:

- Fix wrong tracker links by using existing `trackerURL` function 
- Add edge case control for trackers without `external_id`
- Update snaphots
- Add `trackerURL` function to test mocks
- Handle edge case for empty strings on `trackerURL` function
- Add default values to `osimRuntime` ref

## Considerations:

- In live environments external_ids shouldn't be empty but for local instances of OSIDB and for posibility of corrupted data it was considered to add that edge case control on the FE.
- It wasn't really necessary to add default 'dummy' values to `osimRuntime`  ref, but I considered they are useful as example and that they shouldn't be empty strings under any case. Also considering they are used on: 
[Local Configuration](https://github.com/RedHatProductSecurity/osim/blob/main/docs/overview.md#local-configuration)

Closes OSIDB-3440

### CODE REVIEW

File: src/components/FlawTrackers.vue [hunk 1]
```diff
@@ -302,9 +303,16 @@ function increaseItemsPerPage() {
...
+                <a
+                  v-if="tracker.external_system_id"
+                  :href="trackerUrl(tracker.type, tracker.external_system_id)"
+                  target="_blank"
+                >
                   {{ `${tracker.external_system_id} ` }}<i class="bi-box-arrow-up-right" />
-                </RouterLink>
+                </a>
+                <span v-else title="This tracker doesn't have External ID">
+                  Undefined
```
Comment (MrMarble) [inline src/components/FlawTrackers.vue:314]: I would prefer to use `--` or just empty when there's no ID, `Undefined` seems like a bug as we don't use it anywhere else

File: src/services/TrackerService.ts [hunk 2]
```diff
@@ -93,9 +93,13 @@ export async function getTrackersForFlaws(requestBody: TrackersFilePost) {
 export function trackerUrl(type: string, id: string): string {
   switch (type) {
     case 'BUGZILLA':
-      return (new URL(`/${id}`, osimRuntime.value.backends.bugzilla)).href;
+      return osimRuntime.value.backends.bugzilla
+        ? (new URL(`/${id}`, osimRuntime.value.backends.bugzilla)).href
+        : 'http://bugzilla-service:8001';
```
Comment (MrMarble) [inline src/services/TrackerService.ts:98]: Maybe an _or_ `||` inside the URL function looks better?
```suggestion
      return (new URL(`/${id}`, osimRuntime.value.backends.bugzilla || 'http://bugzilla-service:8001')).href;
```

File: src/services/TrackerService.ts [hunk 3]
```diff
@@ -93,9 +93,13 @@ export async function getTrackersForFlaws(requestBody: TrackersFilePost) {
...
     case 'JIRA':
-      return (new URL(`/browse/${id}`, osimRuntime.value.backends.jiraDisplay)).href;
+      return osimRuntime.value.backends.jiraDisplay
+        ? (new URL(`/browse/${id}`, osimRuntime.value.backends.jiraDisplay)).href
+        : 'http://jira-service:8002';
```
Comment (MrMarble) [inline src/services/TrackerService.ts:102]: also here 

```suggestion
      return (new URL(`/browse/${id}`, osimRuntime.value.backends.jiraDisplay || 'http://jira-service:8002')).href;
```

File: src/stores/osimRuntime.ts [hunk 4]
```diff
@@ -44,10 +44,17 @@ const OsimRuntime = z.object({
...
+  backends: {
+    osidb: 'http://osidb-service:8000',
+    osidbAuth: 'kerberos',
+    bugzilla: 'http://bugzilla-service:8001',
+    jira: 'http://jira-service:8002',
+    errata: 'http://errata-service:8003',
+    jiraDisplay: 'http://jira-service:8002',
+  },
+  osimVersion: { rev: 'dev', tag: 'dev', timestamp: '1970-01-01T00:00:00Z' },
+  error: 'error',
```
Comment (MrMarble) [inline src/stores/osimRuntime.ts:57]: why these changes?

File: src/components/FlawTrackers.vue (hunk 1 above)
Comment (C-Valen) [reply src/components/FlawTrackers.vue:314]: Empty was my first option, but when checking the UI it was feeling weird.

I understand that `undefined` feels like a bug to developers, but I think having empty or random characters can be more confusing to the users than 'undefined', cause it provides direct self-explanatory information.

Not strong preference.

File: src/services/TrackerService.ts (hunk 2 above)
Comment (C-Valen) [reply src/services/TrackerService.ts:98]: Totally +1

File: src/stores/osimRuntime.ts (hunk 4 above)
Comment (C-Valen) [reply src/stores/osimRuntime.ts:57]: It wasn't really necessary to add default 'dummy' values to osimRuntime ref, but I considered they are useful as example and that they shouldn't be empty strings under any case. Also considering they are used on:
[Local Configuration](https://github.com/RedHatProductSecurity/osim/blob/main/docs/overview.md?rgh-link-date=2024-09-16T12%3A41%3A58Z#local-configuration)

File: src/components/FlawTrackers.vue (hunk 1 above)
Comment (MrMarble) [reply src/components/FlawTrackers.vue:314]: Its just weird, we don't use it anywhere and to me it seems like the link is broken or something happened and the value is "Undefined" instead of not having a link, maybe the word "None" is better if you don't like the `--`, other than that I think it looks good so I'll give my approve

**Review by MrMarble — APPROVED** [review]

File: src/components/FlawTrackers.vue (hunk 1 above)
Comment (C-Valen) [reply src/components/FlawTrackers.vue:314]: Let'g go for `None`
